packages/browser-ui/src/inspector/views/Palette.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from "react";
+import IViewProps from "./IViewProps";
+import { colorUninitShapes, 
+  colorUninitText,
+	validColorMapNames,
+	PenroseState } from "@penrose/core";
+
+interface IState {
+	originalState: PenroseState | undefined
+}
+
+class Palette extends React.Component<IViewProps, IState> {
```

Comment:
None of the components in the inspector actually alter `history` in `App`. If you choose to do so, you'll need to pass in a callback `appendHistory` to `Inspector`. In `App`, define `appendHistory` to do:
```ts
const appendHistory = (s: State): void => this.setState({hisotry: [...this.state.history, s]})
```